packages/@aws-cdk/core/lib/validation/report.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * Violation produced by the validation plugin.
+ */
+export interface PolicyViolation {
+  /**
+   * The name of the rule.
+   */
+  readonly ruleName: string;
+
+  /**
+   * The description of the violation.
+   */
+  readonly description: string;
+
+  /**
+   * How to fix the violation.
+   *
+   * @default - no fix is provided
+   */
+  readonly fix?: string;
+
+  /**
+   * The severity of the violation, only used for reporting purposes.
+   * This is useful for helping the user discriminate between warnings,
+   * errors, information, etc.
+   *
+   * @default - no severity
+   */
+  readonly severity?: string;
+
+  /**
+   * Additional metadata to include with the rule results.
+   * This can be used to provide additional information that is
+   * plugin specific. The data provided here will be rendered as is.
+   *
+   * @default - no rule metadata
+   */
+  readonly ruleMetadata?: { readonly [key: string]: string }
+}
+
+/**
+ * Resource violating a specific rule.
+ */
+export interface PolicyViolatingResource {
+  /**
+   * The logical ID of the resource in the CloudFormation template.
+   */
+  readonly resourceLogicalId: string;
+
+  /**
+   * The locations in the CloudFormation template that pose the violations.
+   */
+  readonly locations: string[];
+
+  /**
+   * The path to the CloudFormation template that contains this resource
+   */
+  readonly templatePath: string;
+}
+
+/**
+ * Validation produced by the validation plugin, in CFN resource terms
+ */
+export interface PolicyViolationResourceAware extends PolicyViolation {
```

Comment:
We need this because the plugin will report back the `PolicyViolationResourceAware` information, but then the report will use the `PolicyViolationConstructAware`.

Reply:
Right, but does that mean we can't put `violatingResources ` inside `PolicyValidation` and remove this extension? I don't see these being used separately, its always `PolicyViolationResourceAware`

Reply:
oh I see. That means we'll return the `violatingResources` in the json report, which I guess isn't a big deal.

Reply:
Updated.